src/wp-includes/block-patterns.php

```diff
@@ -363,130 +458,94 @@ function _register_theme_block_patterns() {
-								$pattern_data['slug']
-							),
-							'6.0.0'
-						);
-					}
-
-					if ( WP_Block_Patterns_Registry::get_instance()->is_registered( $pattern_data['slug'] ) ) {
-						continue;
-					}
-
-					// Title is a required property.
-					if ( ! $pattern_data['title'] ) {
-						_doing_it_wrong(
-							'_register_theme_block_patterns',
-							sprintf(
-								/* translators: %1s: file name; %2s: slug value found. */
-								__( 'Could not register file "%s" as a block pattern ("Title" field missing)' ),
-								$file
-							),
-							'6.0.0'
-						);
-						continue;
-					}
-
-					// For properties of type array, parse data as comma-separated.
-					foreach ( array( 'categories', 'keywords', 'blockTypes', 'postTypes', 'templateTypes' ) as $property ) {
-						if ( ! empty( $pattern_data[ $property ] ) ) {
-							$pattern_data[ $property ] = array_filter(
-								preg_split(
-									'/[\s,]+/',
-									(string) $pattern_data[ $property ]
-								)
-							);
-						} else {
-							unset( $pattern_data[ $property ] );
-						}
-					}
-
-					// Parse properties of type int.
-					foreach ( array( 'viewportWidth' ) as $property ) {
-						if ( ! empty( $pattern_data[ $property ] ) ) {
-							$pattern_data[ $property ] = (int) $pattern_data[ $property ];
-						} else {
-							unset( $pattern_data[ $property ] );
-						}
-					}
-
-					// Parse properties of type bool.
-					foreach ( array( 'inserter' ) as $property ) {
-						if ( ! empty( $pattern_data[ $property ] ) ) {
-							$pattern_data[ $property ] = in_array(
-								strtolower( $pattern_data[ $property ] ),
-								array( 'yes', 'true' ),
-								true
-							);
-						} else {
-							unset( $pattern_data[ $property ] );
-						}
-					}
-
-					// Translate the pattern metadata.
-					$text_domain = $theme->get( 'TextDomain' );
-					// phpcs:ignore WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralDomain,WordPress.WP.I18n.LowLevelTranslationFunction
-					$pattern_data['title'] = translate_with_gettext_context( $pattern_data['title'], 'Pattern title', $text_domain );
-					if ( ! empty( $pattern_data['description'] ) ) {
-						// phpcs:ignore WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralDomain,WordPress.WP.I18n.LowLevelTranslationFunction
-						$pattern_data['description'] = translate_with_gettext_context( $pattern_data['description'], 'Pattern description', $text_domain );
-					}
-
-					// The actual pattern content is the output of the file.
-					ob_start();
-					include $file;
-					$pattern_data['content'] = ob_get_clean();
-					if ( ! $pattern_data['content'] ) {
-						continue;
-					}
-
-					register_block_pattern( $pattern_data['slug'], $pattern_data );
-				}
+
+		if ( ! preg_match( '/^[A-z0-9\/_-]+$/', $pattern['slug'] ) ) {
+			_doing_it_wrong(
+				__FUNCTION__,
+				sprintf(
+					/* translators: %1s: file name; %2s: slug value found. */
```

Comment:
This should be formatted like other such translator comments in core:
```suggestion
					/* translators: 1: file name, 2: slug value found */
```